tests/test_api_long.py

```diff
@@ -22,29 +29,31 @@ def test_automated_find(self):
             if entity_type in ("Asset", "Task", "Shot", "Attachment",
                                "Candidate"):
                 continue
-            print "Finding entity type", entity_type
+            print("Finding entity type %s" % entity_type)
 
             fields = self.sg.schema_field_read(entity_type)
             if not fields:
-                print "No fields for %s skipping" % (entity_type,)
+                print("No fields for %s skipping" % entity_type)
                 continue
 
             # trying to use some different code paths to the other find test
-            # pivot_column fields aren't valid for sorting so ensure we're 
+            # pivot_column fields aren't valid for sorting so ensure we're
             # not using one.
             order_field = None
-            for field_name, field in fields.iteritems():
-                if field['data_type']["value"] != 'pivot_column':
+            for field_name, field in six.iteritems(fields):
+                if field['data_type']["value"] in allow_sorting_types:
```

Comment:
Added a comment in the code as well, but essentially this was dependant on the dictionary order and began failing in python 3 because the dictionary order changed and new keys were used for sorting.  It turned out that in some cases these were not sortable keys.  Since `no_sorting` is not exposed, after some discussion with Benjamin and Brandon this seemed like the cleanest option.